datafusion/physical-plan/src/insert.rs

```diff
@@ -224,8 +224,16 @@ impl ExecutionPlan for DataSinkExec {
         if partition != 0 {
             return internal_err!("DataSinkExec can only be called on partition 0!");
         }
+        let input = Arc::clone(&self.input);
```

Comment:
I'm not really sure that the number of input rows will always match the number of inserted rows. What happens if some rows aren't valid? Does the insert query fail with 0 affected rows?

Reply:
> I'm not really sure that the number of input rows will always match the number of inserted rows. What happens if some rows aren't valid? Does the insert query fail with 0 affected rows?

Probably that can happen in non transactional database, like ours. I don't know how exactly reproduce such scenario. I'll try.